tests/metrics/utils/test_prompy_template.py

```diff
@@ -0,0 +1,18 @@
+import pytest
+
+from mlflow.metrics.utils.prompt_template import PromptTemplate
+
+
+def test_prompt_template_formatting():
+    prompt = PromptTemplate(template_str="Say {foo}")
+    assert prompt.format_prompt(foo="bar") == "Say bar"
+
+    prompt = PromptTemplate(template_str="Say {foo} {baz}")
+    with pytest.raises(KeyError, match="baz"):
+        prompt.format_prompt(foo="bar")
```

Comment:
i wonder if we should catch this and return a more helpful MlflowException exception instead.